components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts

```diff
@@ -0,0 +1,380 @@
+  const ancestor = elem.closest('.commit')
+  if (ancestor) {
+    const avatarCollection = ancestor.getElementsByClassName('avatar')
+    if (avatarCollection.length) {
+      const avatar = avatarCollection[0] as HTMLElement
+      const avatarChildren = avatar.children
+      if (avatarChildren.length) {
+        const userName = (avatarChildren[0] as HTMLImageElement).alt.slice(1)
+        return {
+          mediaType: 'github',
+          userName: userName || ''
+        }
+      }
+    }
+  }
+  return null
+}
+
+const commitLinksInsertFunction = (parent: Element) => {
+  const tipElem = createBraveTipAction(parent, getCommitLinksMetaData)
+  tipElem.style.marginLeft = '9px'
+  parent.appendChild(tipElem)
+}
+
+const getStarringContainerMetaData = (elem: Element): RewardsTip.MediaMetaData | null => {
+  const ancestor = elem.closest('.d-block')
+  if (ancestor) {
+    const aTags = ancestor.getElementsByTagName('A')
+    if (aTags.length) {
+      const aTag = aTags[0]
+      const userName = (aTag as HTMLAnchorElement).href.split('/')[3]
+      return {
+        mediaType: 'github',
+        userName: userName || ''
+      }
+    }
+  }
+  return null
+}
+
+const starringContainerInsertFunction = (parent: Element) => {
+  if (isBlackListTab(window.location.search)) {
+    return
+  }
+  const correctSubclass = 'starring-container'
+  if (!parent.getElementsByClassName(correctSubclass).length) {
+    return
+  }
+  const subElem = parent.getElementsByClassName(correctSubclass)[0]
+  const tipElem = createBraveTipAction(parent, getStarringContainerMetaData)
+  tipElem.classList.add('d-inline-block')
+  tipElem.style.minWidth = '60px'
+  parent.insertBefore(tipElem, subElem)
+}
+
+const getPageHeadMetaData = (elem: Element): RewardsTip.MediaMetaData | null => {
+  const elems = document.getElementsByClassName('gisthead')
+  if (elems.length) {
+    elem = elems[0]
+    const authors = elem.getElementsByClassName('author')
+    if (authors.length) {
+      const author = authors[0]
+      const aTags = author.getElementsByTagName('A')
+      if (aTags.length) {
+        const aTag = aTags[0] as HTMLAnchorElement
+        const userName = aTag.innerHTML
+        return {
+          mediaType: 'github',
+          userName: userName || ''
+        }
+      }
+    }
+  }
+  return null
+}
+
+const pageheadInsertFunction = (parent: Element) => {
+  const subdomain = window.location.host.split('.')[0]
+  if (subdomain === 'gist' &&
+      window.location.pathname.slice(1).split('/').length > 1) {
+    const tipElem = createBraveTipAction(parent, getPageHeadMetaData)
+    parent.appendChild(tipElem)
+  }
+}
+
+const getMemberListItemMetaData = (elem: Element): RewardsTip.MediaMetaData | null => {
+  const ancestor = elem.closest('.table-list-cell')
+  if (ancestor) {
+    const aTags = ancestor.getElementsByTagName('A')
+    if (aTags.length) {
+      const aTag = aTags[0] as HTMLAnchorElement
+      const userName = aTag.href.split('/')[3]
```

Comment:
check if href is undefined and if you have array that has 4 items inside

Reply:
fixed